### PR TITLE
fix: trigger copa-action release on new copa release

### LINKED ISSUE #266: [BUG] fix image publishing for copa-action
### Version of copa

_No response_

### Expected Behavior

_No response_

### Actual Behavior

release process fails with the following error: 
```
#10 ERROR: failed to push ghcr.io/project-copacetic/copa-action:v0.4.0: failed to authorize: failed to fetch anonymous token: unexpected status from GET request to https://ghcr.io/token?scope=repository%3Aproject-copacetic%2Fcopa-action%3Apull%2Cpush&service=ghcr.io: 403 Forbidden
```

Looks like action needs permission to push to GHCR

### Steps To Reproduce

https://github.com/project-copacetic/copacetic/actions/runs/5942389504/job/16115269547

### PR DESCRIPTION
**Pull request by ashnamehrotra**
Creates request to trigger copa-action release event whenever there is a new release of copa. 

Related copa action PR: https://github.com/project-copacetic/copa-action/pull/14

Describe the changes in this pull request using active verbs such as _Add_, _Remove_, _Replace_ ...

Closes #266

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -37,16 +37,12 @@ jobs:
...
-
-      - name: Build and push copa-action image with new version
+      - name: Trigger Repo Dispatch for Copa Action Release
         run: |
-          tag="$(echo "${{ github.ref }}" | tr -d 'refs/tags/v')"
-          docker buildx build --build-arg copa_version=${tag} -t ghcr.io/project-copacetic/copa-action:v"$tag" --push .
+            tag="$(echo "${{ github.ref }}" | tr -d 'refs/tags/v')"
+            payload="{\"event_type\": \"copa_release\", \"client_payload\": { \"version\": \"$tag\" }}"
+            curl -H "Accept: application/vnd.github.everest-preview+json" \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
```
Comment (sozercan) [inline .github/workflows/release.yml:45]: how does this token work to trigger copa-action?

Comment (ashnamehrotra) [reply .github/workflows/release.yml:45]: I tested it with this token on these runs: https://github.com/ashnamehrotra/copacetic/actions/runs/5969797230/job/16196305199 & https://github.com/ashnamehrotra/copa-action/actions/runs/5969828753/job/16196388231

But this was set as a personal access token so it gave access to both repos. Is there a way to scope it to the project-copacetic org? I'm not sure if I have access to see the tokens right now

Comment (sozercan) [reply .github/workflows/release.yml:45]: right, you can do that with a PAT but not with repo specific token. I am not sure if you can access cross-repos with a token from a repo

Comment (ashnamehrotra) [reply .github/workflows/release.yml:45]: I wasn't able to figure out a good way to do this without access issues. For now I updated https://github.com/project-copacetic/copa-action/pull/14 to release manually. Another way to do this automatically could be to use [Github Apps](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow), but I am not sure if it is worth creating an app for the authentication?

**Pull request closed without merging**